### PR TITLE
Fix to #27229 - Test: Convert query tests to conditional theory

### LINKED ISSUE #27229: Test: Convert query tests to conditional theory
https://github.com/dotnet/efcore/blob/e42407cb9332bc9bb4357342bf2ddd312873cb23/test/EFCore.Cosmos.FunctionalTests/Query/NorthwindSelectQueryCosmosTest.cs#L319-L330

Test is of type void. Method used inside is AssertTranslationFailed is async and not awaited. There are several tests like that in the same file. Could be more.

### PR DESCRIPTION
**Pull request by maumar**
Converting Northwind query tests to query infra.

Fixes #27229

### CODE REVIEW

File: test/EFCore.Cosmos.FunctionalTests/Query/NorthwindMiscellaneousQueryCosmosTest.cs [hunk 1]
```diff
@@ -3033,12 +2955,9 @@ public override async Task OrderBy_Dto_projection_skip_take(bool async)
     public override void Streaming_chained_sync_query()
     {
         // Cross collection join. Issue #17246.
-        AssertTranslationFailed(
-            () =>
-            {
-                base.Streaming_chained_sync_query();
-                return Task.CompletedTask;
-            });
+        Assert.Contains(
```
Comment (maumar) [inline test/EFCore.Cosmos.FunctionalTests/Query/NorthwindMiscellaneousQueryCosmosTest.cs:2958]: explicitly sync query

Comment (smitpatel) [reply test/EFCore.Cosmos.FunctionalTests/Query/NorthwindMiscellaneousQueryCosmosTest.cs:2958]: Can convert to theory and replace last call to ToList with a ternary based on async switch. Rename test to streaming chained query.
The original intent of test was far gone with removal of client eval, we can now test multiple chained queries running.

**Comment by maumar** [comment]
@smitpatel updated

**Review by ajcvickers — APPROVED** [review]

File: test/EFCore.Specification.Tests/Query/NorthwindMiscellaneousQueryTestBase.cs [hunk 2]
```diff
@@ -214,24 +239,36 @@ var results
...
     {
         using var context = CreateContext();
         var variableName = "test";
         var differentVariableName = "test";
 
-        context.Set<Customer>().Where(e => e.CustomerID == "ALFKI")
-            .Where(e2 => InMemoryCheck.Check(variableName, e2.CustomerID) || true).Count();
+        if (async)
+        {
+            await context.Set<Customer>().Where(e => e.CustomerID == "ALFKI")
```
Comment (smitpatel) [inline test/EFCore.Specification.Tests/Query/NorthwindMiscellaneousQueryTestBase.cs:252]: generally easier to maintain defining queryables outside of ifs and then inside if just call terminating operator over queryable

File: test/EFCore.Sqlite.FunctionalTests/Query/NorthwindMiscellaneousQuerySqliteTest.cs [hunk 3]
```diff
@@ -350,6 +350,18 @@ public override async Task Client_code_unknown_method(bool async)
...
+    public override async Task Entity_equality_through_subquery_composite_key(bool async)
+    {
+        var message = (await Assert.ThrowsAsync<InvalidOperationException>(
+            () => base.Entity_equality_through_subquery_composite_key(async))).Message;
+
+        Assert.Equal(
+            CoreStrings.EntityEqualityOnCompositeKeyEntitySubqueryNotSupported("==", nameof(OrderDetail)),
+            message);
+
+        AssertSql();
```
Comment (smitpatel) [inline test/EFCore.Sqlite.FunctionalTests/Query/NorthwindMiscellaneousQuerySqliteTest.cs:362]: remove

Comment (smitpatel) [reply test/EFCore.Sqlite.FunctionalTests/Query/NorthwindMiscellaneousQuerySqliteTest.cs:362]: Also in other tests here unless we believe there is a reason to verify SQL

Comment (maumar) [reply test/EFCore.Sqlite.FunctionalTests/Query/NorthwindMiscellaneousQuerySqliteTest.cs:362]: with "the no tests left behind" changes we now always check sql, even when the test throws an exception

**Pull request merged**